models/BaseORMService.cfc

```diff
@@ -1943,6 +1929,52 @@ process(
 	/********************************* PRIVATE METHODS **************************************/
 	/*****************************************************************************************/
 
+	/**
+	 * Get property values for the given entity.
+	 *
+	 * @ormSession the current ORM session. Will (probably) throw an exception if session is not open.
+	 * @entity The entity to retrieve property values on.
+	 * 
+	 * @see https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/persister/entity/EntityPersister.html#getPropertyValues-java.lang.Object-
+	 */
+	private function getPropertyValues( required ormSession, required entity ){
+		var hibernateMD = getEntityMetadata( arguments.entity );
+		if ( val( left( variables.ORM.getHibernateVersion(), 3 ) ) < 4.0 ){
+			return hibernateMD.getPropertyValues(
+				arguments.entity,
+				variables.ORM.getSessionEntityMode( arguments.ormSession, arguments.entity )
+			);
+		} else {
+			return hibernateMD.getPropertyValues( arguments.entity );
+		}
+	}
+
+	/**
+	 * For Hibernate 5.4+, convert legacy-style positional parameters to JPA-syntax.
+	 * Uses Java regex library since neither Lucee nor ACF offer modern regex support.
+	 *
+	 * @hql Hibernate Query Language string to make safe for Hibernate v5.4+.
+	 * @see https://hibernate.atlassian.net/browse/HHH-12101
+	 */
+	private string function getHibernateSafeHQL( required string hql ){
```

Comment:
I am not sure how I feel about this. We talked about it, but not sure on performance and the overhead of doing it this way. Thoughts?

Reply:
Honestly, I think this has the potential for causing more headaches. I personally say we should bump the major version and make devs upgrade their HQL to the JPA syntax. I could still release something like the above :point_up: as a conversion tool, like cfscript.me.

Reply:
@lmajano I removed this "feature" as discussed. We can show it in the docs or even wrap it into a "HibernateUtils" ColdBox extension.